### PR TITLE
Remove assert statements from code

### PR DESCRIPTION
**Pull request by ashwinvin**
This pr aims to remove assert statements and replace them with respective exceptions. 

## Reason
Asserts don't get run when python is started with `-OO` flag.

### CODE REVIEW

File: starlite/handlers.py [hunk 1]
```diff
@@ -68,7 +68,8 @@ def resolve_dependencies(self) -> Dict[str, Provide]:
         """
         Returns all dependencies correlating to handler function's kwargs that exist in the handler's scope
         """
-        assert self.signature_model, "resolve_dependencies cannot be called before a signature model has been generated"
+        if not self.signature_model:
+            raise RuntimeError("resolve_dependencies cannot be called before a signature model has been generated")
```
Comment (Goldziher) [inline starlite/handlers.py:72]: I'm not sure RuntimeError is correct here... but I have no better idea.

File: starlite/handlers.py [hunk 2]
```diff
@@ -346,14 +350,17 @@ def validate_handler_function(self) -> None:
         signature_model = cast(BaseModel, self.signature_model)
         return_annotation = Signature.from_callable(cast(AnyCallable, self.fn)).return_annotation
 
-        assert return_annotation is None, "websocket handler functions should return 'None' values"
-        assert "socket" in signature_model.__fields__, "websocket handlers must set a 'socket' kwarg"
+        if return_annotation is not None:
+            raise TypeError("websocket handler functions should return 'None'")
+        if "socket" not in signature_model.__fields__:
+            raise AttributeError("websocket handlers must set a 'socket' kwarg")
```
Comment (Goldziher) [inline starlite/handlers.py:356]: ImproperlyConfiguredException

File: starlite/handlers.py [hunk 3]
```diff
@@ -346,14 +350,17 @@ def validate_handler_function(self) -> None:
...
 
     async def handle_websocket(self, web_socket: WebSocket) -> None:
         """
         Handles a given Websocket in relation to self.
         """
-        assert self.fn, "cannot call a route handler without a decorated function"
+        if not self.fn:
+            raise AttributeError("cannot call a route handler without a decorated function")
```
Comment (Goldziher) [inline starlite/handlers.py:363]: same

File: starlite/handlers.py (hunk 1 above)
Comment (ashwinvin) [reply starlite/handlers.py:72]: Can you explain how this exception gets triggered? :slightly_smiling_face:

File: starlite/handlers.py [hunk 4]
```diff
@@ -98,7 +99,8 @@ def validate_handler_function(self) -> None:
         """
         Validates the route handler function once it's set by inspecting its return annotations
         """
-        assert self.fn, "cannot call validate_handler_function without first setting self.fn"
+        if not self.fn:
+            raise AttributeError("cannot call validate_handler_function without first setting self.fn")
```
[the remark this answers is not in the document]
Comment (ashwinvin) [reply starlite/handlers.py:103]: :+1:

File: starlite/handlers.py (hunk 1 above)
Comment (Goldziher) [reply starlite/handlers.py:72]: What do you mean?

Comment (ashwinvin) [reply starlite/handlers.py:72]: nvm I figured it out :+1:

**Pull request merged**